### PR TITLE
Problem: (CRO-591) HD wallet module is not re-usable

### PR DESCRIPTION
**Pull request by calvinlauco**
Solution: Extract `HdKey` into `HDSeed` and `Mnemonic`

### CODE REVIEW

**Comment by calvinaco** [comment]
bors try

File: client-core/src/hd_seed.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+impl From<&Seed> for HDSeed {
+    fn from(seed: &Seed) -> Self {
+        HDSeed {
+            bytes: seed.as_bytes().to_vec(),
+        }
+    }
+}
+
+impl HDSeed {
+    /// Generate new mnemonic words and create new HD seed from it
```
Comment (calvinlauco) [inline client-core/src/hd_seed.rs]: The API doc is outdated.

File: client-core/src/hd_seed.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    }
+}
+
+impl From<&Seed> for HDSeed {
+    fn from(seed: &Seed) -> Self {
+        HDSeed {
+            bytes: seed.as_bytes().to_vec(),
+        }
+    }
+}
```
Comment (calvinlauco) [inline client-core/src/hd_seed.rs]: This From trait is not used anywhere

File: client-core/src/mnemonic.rs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            #[inline]
+            fn visit_str<E>(self, value: &str) -> std::result::Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                Mnemonic::from_str(value).map_err(|e| E::custom(format!("{}", e)))
+            }
+        }
+
+        deserializer.deserialize_str(StrVisitor)
```
Comment (yihuang) [inline client-core/src/mnemonic.rs]: Seems simpler to write as ``Mnemonic::from_str(&String::deserialize(deserializer)?).map_err(...)?``.

Comment (calvinlauco) [reply client-core/src/mnemonic.rs]: Thanks! I have updated the code and added test cases

**Comment by calvinlauco** [comment]
> looks ok -- can you squash the commits?

Done, and I have changed the serialization code for Mnemonic after Huang Yi feedback.

**Pull request merged**